### ISSUE TITLE
import_ted.py causing memory issue

### ISSUE THREAD

**Issue by michaelklachko** [mozilla/DeepSpeech#1477]
- **Have I written custom code (as opposed to running examples on an unmodified clone of the repository)**: NO
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux Ubuntu 16.04
- **TensorFlow installed from (our builds, or upstream TensorFlow)**: Upstream
- **TensorFlow version (use command below)**: v1.6.0-0-gd2e24b6039
- **Python version**: 3.6.6
- **Bazel version (if compiling from source)**: N/A
- **GCC/Compiler version (if compiling from source)**: N/A
- **CUDA/cuDNN version**: 9.0/7.0
- **GPU model and memory**: 4xTitanX Pascal
- **Exact command to reproduce**: python bin/import_ted.py data/TD

I'm getting the following error:

```
Traceback (most recent call last):
  File "bin/import_ted.py", line 153, in <module>
    _download_and_preprocess_data(sys.argv[1])
  File "bin/import_ted.py", line 27, in _download_and_preprocess_data
    local_file = base.maybe_download(TED_DATA, data_dir, TED_DATA_URL)
  File "/home/michael/miniconda2/envs/moz/lib/python3.6/site-packages/tensorflow/contrib/learn/python/learn/datasets/base.py", line 201, in maybe_download
    temp_file_name, _ = urlretrieve_with_retry(source_url)
  File "/home/michael/miniconda2/envs/moz/lib/python3.6/site-packages/tensorflow/contrib/learn/python/learn/datasets/base.py", line 156, in wrapped_fn
    return fn(*args, **kwargs)
  File "/home/michael/miniconda2/envs/moz/lib/python3.6/site-packages/tensorflow/contrib/learn/python/learn/datasets/base.py", line 183, in urlretrieve_with_retry
    return urllib.request.urlretrieve(url, filename)
  File "/home/michael/miniconda2/envs/moz/lib/python3.6/urllib/request.py", line 281, in urlretrieve
    tfp.write(block)
  File "/home/michael/miniconda2/envs/moz/lib/python3.6/tempfile.py", line 485, in func_wrapper
    return func(*args, **kwargs)
OSError: [Errno 28] No space left on device

(asr) michael@Pascal:~$ df -h
Filesystem                     Size  Used Avail Use% Mounted on
udev                            32G     0   32G   0% /dev
tmpfs                          6.3G  9.5M  6.3G   1% /run
/dev/nvme0n1p4                 930G  712G  171G  81% /
tmpfs                           32G     0   32G   0% /dev/shm
tmpfs                          5.0M     0  5.0M   0% /run/lock
tmpfs                           32G     0   32G   0% /sys/fs/cgroup
**tmpfs                           32G   32G     0 100% /tmp**
/dev/sda1                      1.8T  214G  1.5T  13% /storage
cgmfs                          100K     0  100K   0% /run/cgmanager/fs
tmpfs                          6.3G     0  6.3G   0% /run/user/1001
```

My machine has 64GB of RAM, and enough disk space (nvme0n1p4). The first time this happened, I rebooted the machine, and tried again: same issue.

Any hints on how to troubleshoot this are appreciated! 

BTW, https://lium.univ-lemans.fr/en/ted-lium3 is out, so it might be a good idea to update the script (release 3 doubled the size of the training dataset).

**Comment by lissyx** [comment]
@michaelklachko Your `/tmp` is full.

**Comment by kdavis-mozilla** [comment]
Looks like its running out of disk space.

My guess is that it's writing files in `/tmp` that are then being removed when the process stops. So you don't see them in the `df -h` above.

What might be useful is to monitor disk space as the import is occurring. In other words start the import as you did, then in a second terminal run
```bash
(asr) michael@Pascal:~$ watch df -h
```
and monitor how the disk usage evolves.

**Comment by lissyx** [comment]
@michaelklachko Maybe you could override with `TMP` env var to somewhere with more free space than your `/tmp` ?

**Comment by michaelklachko** [comment]
Here's what /tmp looked like when the process crashed:

```
(asr) michael@Pascal:~$ ll /tmp
total 32G
drwx------ 3 root     60 Jul 26 14:38 systemd-private-aa3a127ac3fe4daab7bc808d91b2afe6-systemd-timesyncd.service-p2u5uU
-rw------- 1 root    29K Jul 26 14:57 nv-dZpLpl
-rw------- 1 root    29K Jul 26 14:42 nv-Ww4slB
-rw------- 1 root    139 Jul 26 15:09 template-nEZtJ0
-rw------- 1 root    651 Jul 26 14:57 template-VQDV6X
-rw------- 1 michael 32G Jul 26 19:09 tmppthivwfd
```

Can we perhaps try to fix the script, rather than my machine? Why is it writing to /tmp? Ideally it should not consume much memory, just download/extract/write to the directory I specified as argument...

**Comment by lissyx** [comment]
@michaelklachko The script uses TensorFlow's tooling for downloading, so if there's a bug, it's on their side.

**Comment by lissyx** [comment]
https://github.com/tensorflow/tensorflow/blob/r1.6/tensorflow/contrib/learn/python/learn/datasets/base.py#L186

**Comment by michaelklachko** [comment]
@lissyx I used 3 other import scripts (CV, LS, and VoxForge) without any issues. Something must be different about this one...

**Comment by lissyx** [comment]
@michaelklachko Well, they all use `base.maybe_download` from TensorFlow, so I don't know what might be different. Obviously, you have a stale file `tmppthivwfd`, but I'd suspect this comes from your first failure. Could it be that you got unlucky and something else consumed enough space in `/tmp` during your download, to make it fail?

**Comment by lissyx** [comment]
BTW, looking at the stale file name and the stack, it's clear that the failure occuped while downloading.

**Comment by michaelklachko** [comment]
Ok, I cleared /tmp, let me try again.

**Comment by lissyx** [comment]
@michaelklachko I don't remember the size of that dataset, but if it's above 32GB, you might still want to `TMP=...` when running the import script to use some place with more space.

**Comment by lissyx** [comment]
@michaelklachko Would you mind making that a PR ? You've already made nice progress on that, and this would kill #1531. I guess the rationale for this tensorflow download is just we used it without thinking back then, and that's it.
